mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -141,6 +135,48 @@ def __init__(self, name=None, bus=None, use_settings=True):
         self.bind(bus)
         #: Mycroft global configuration. (dict)
         self.config_core = Configuration.get()
+
+        self.settings = None
+
+        #: To not break existing setups,
+        #: save to skill directory if the file exists already
+        self.settings_write_path = Path(self.root_dir)
+
+        #: Otherwise save to XDG_CONFIG_DIR
+        if not self.settings_write_path.joinpath('settings.json').exists():
+            self.settings_write_path = Path(BaseDirectory.save_config_path(
+                'mycroft', 'skills', self.name))
+
+        self.settings_read_path = None
+
+        #: To note break existing setups,
+        #: read from skill directory if the settings file exists there
+        self.settings_read_path = Path(self.root_dir)
+
+        #: Then, check XDG_CONFIG_DIR
+        if not self.settings_read_path.joinpath('settings.json').exists():
+            for dir in BaseDirectory.load_config_paths('mycroft',
+                                                       'skills',
+                                                       self.name):
+                path = Path(dir)
+                #: If there is a settings file here, use it
+                if path.joinpath('settings.json').exists():
+                    self.settings_read_path = path
+                    break
+
+        #: Lastly, check /etc/mycroft
+        if not self.settings_read_path.joinpath('settings.json').exists():
+            self.settings_read_path = Path(
+                '/etc/mycroft/skills/').joinpath(self.name)
+
+        if use_settings:
+            self._init_settings()
+
+        #: Set to register a callback method that will be called every time
+        #: the skills settings are updated. The referenced method should
+        #: include any logic needed to handle the updated settings.
+        self.settings_change_callback = None
+
```

Comment:
This whole block is quite big perhaps move it into the _init_settings() and pass the use_settings as a parameter